### PR TITLE
add dynamic toolbar buttons + HowTo

### PR DESCRIPTION
**Pull request by pmario**
related to:  Proposal: Enable Toolbar Buttons to use dynamic disabled status #5281 

If you want to see, how it works "play with the HowTo". ... The tricky part is: Disabling the shortcut, if the button is visible _and_ disabled. 

Play with it at: https://pmario.github.io/kitchensink/5249-dynamic-toolbar-buttons

See the "in code" comment

### CODE REVIEW

File: core/ui/EditTemplate/body-toolbar-button.tid [hunk 1]
```diff
@@ -35,22 +35,23 @@ title: $:/core/ui/EditTemplate/body/toolbar/button
...
 
   class="tc-btn-invisible $(buttonClasses)$"
   tooltip=<<tooltip-text>>
   actions={{!!actions}}
+  disabled=<<disabled>>
 
 ><span
 
-  data-tw-keyboard-shortcut={{!!shortcuts}}
+  data-tw-keyboard-shortcut={{{ [<disabled>compare:string:eq[yes]else{!!shortcuts}] +[trim[yes]] }}}
```
Comment (pmario) [inline core/ui/EditTemplate/body-toolbar-button.tid:47]: This is tricky: ... 

 - `[<disabled>` .. gets the info from `$set` in line 38
 - `compare:string:eq[yes]` ... returns "yes" if the button should be disabled
   - if the tiddler or the field doesn't exist ... I till follow the "else" path
 - `else[!!shortcuts]` .. will set the data-tw... attribute, which is needed for the keyboard handler
 - `+[trim[yes]]` is needed to return an empty stirng, if the shortcut is disabled

Comment (saqimtiaz) [reply core/ui/EditTemplate/body-toolbar-button.tid:47]: Just took a quick glance, but would this work, and if so be more legible?

`[<disabled>match[yes]then[]else{!!shortcuts}]`

Comment (pmario) [reply core/ui/EditTemplate/body-toolbar-button.tid:47]: Seems to work. Thx for the input!

**Comment by pmario** [comment]
@Jermolene ... What do you think about it? 

 - It contains the code to create dynamic toolbar buttons and 
 - an example how-to create toolbar buttons and keyboard shortcuts. 

We don't have "hands on" docs like that atm.

**Comment by Jermolene** [comment]
> @Jermolene ... What do you think about it?
> 
> * It contains the code to create dynamic toolbar buttons and
> * an example how-to create toolbar buttons and keyboard shortcuts.
> 
> We don't have "hands on" docs like that atm.

I think I commented on the original ticket that it was a good idea. We can consider this for post-v5.1.23, if you'd like to add it to the list.

**Comment by pmario** [comment]
@Jermolene ... bump

**Comment by pmario** [comment]
@Jermolene bump

Play with it at: https://pmario.github.io/kitchensink/5249-dynamic-toolbar-buttons

**Comment by twMat** [comment]
Looks nice but it is not clear in the demo where such a button would be located? (In the sidebar>tools? In the editor toolbar?)

**Comment by pmario** [comment]
It's an editor toolbar named "temporary Bold" So it's at the end of the row with the same function as the "bold" button.

**Comment by twMat** [comment]
No, I mean where does the GUI for your mechanims show up? How/where does the user access your feature? Sorry if I misunderstand something.

**Comment by Jermolene** [comment]
Hi @pmario this PR features lots of whitespace changes. To be mergeable, the changes in the PR need to be restricted to functional changes. I'm happy to accept a separate PR that fixes whitespace issues.

**Comment by pmario** [comment]
@Jermolene ... Did remove the whitespace

**Comment by pmario** [comment]
>No, I mean where does the GUI for your mechanims show up? How/where does the user access your feature? Sorry if I misunderstand something.

It's a "How to create dynamic **editor toolbar** buttons" ... So the button shows up in the editor. 

It also explains, how to create shortcuts for new toolbar buttons. 

And it adds a new feature to every existing toolbar button and every new toolbar button. .. It allows users to use the "disabled" parameter, that was introduced in 5.1.23

**Comment by Jermolene** [comment]
Thanks @pmario

**Pull request merged**